Zend/zend_inheritance.c

```diff
@@ -570,6 +570,19 @@ static inheritance_status zend_is_class_subtype_of_type(
 	if (have_unresolved) {
 		return INHERITANCE_UNRESOLVED;
 	}
+
+	/* Replacing static with self in final classes is okay */
+	if ((ZEND_TYPE_FULL_MASK(proto_type) & MAY_BE_STATIC) &&
+		(fe_scope->ce_flags & ZEND_ACC_FINAL) && unlinked_instanceof(fe_scope, proto_scope)) {
```

Comment:
Do we need the `unlinked_instanceof(fe_scope, proto_scope)`? Removing it doesn't make any tests fail. FWIU, if you remove the `return INHERITANCE_ERROR;` this can also be removed.